python/python/res/simulator/batch_simulator.py

```diff
@@ -107,20 +120,25 @@ def start(self, case_name, controls):
             assert isinstance(geo_id, int)
 
             node_id = NodeId( 0, sim_id)
-            if len(control_dict) != len(self.control_keys):
-                raise ValueError("Not all keys supplied in controls")
-
-            for key in control_dict.keys():
-               config_node = ens_config[key]
-               ext_config = config_node.getModelConfig( )
-               values = control_dict[key]
-               if not len(values) == len(ext_config):
-                   raise ValueError("Wrong number of values for:%s" % key)
-
-               node = EnkfNode(config_node)
-               ext_node = node.as_ext_param( )
-               ext_node.set_vector(values)
-               node.save(fs, node_id)
+            if set(control_dict.keys()) != set(self.controls_format.keys()):
+                err_msg = "Mismatch between initialized and provided control names."
+                raise KeyError(err_msg)
+
+            for control_name, control in control_dict.iteritems():
+                if set(control.keys()) != set(self.controls_format[control_name]):
```

Comment:
I am not too happy about this duplication of logic/configuration of keys, the `ext_param` implementation is the authorative source regarding these keys. The 
```Python
for var_name, value in control.iteritems():     
    ext_node[var_name] = value
```
will already raise a `KeyError` for invalid key. Could in addition have:
```Python
if len(ext_config) != len(control):
   raise ValueError("Not all keys set")
```
alternatively a convenience method could be added here: https://github.com/Statoil/libres/blob/master/python/python/res/enkf/config/ext_param_config.py

Reply:
I completely agree. I've fixed it now.